clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -0,0 +1,375 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file;
+
+import java.io.IOException;
+import java.util.List;
+
+import tachyon.TachyonURI;
+import tachyon.annotation.PublicApi;
+import tachyon.client.file.options.CreateDirectoryOptions;
+import tachyon.client.file.options.CreateFileOptions;
+import tachyon.client.file.options.DeleteOptions;
+import tachyon.client.file.options.ExistsOptions;
+import tachyon.client.file.options.FreeOptions;
+import tachyon.client.file.options.GetStatusOptions;
+import tachyon.client.file.options.ListStatusOptions;
+import tachyon.client.file.options.LoadMetadataOptions;
+import tachyon.client.file.options.MountOptions;
+import tachyon.client.file.options.OpenFileOptions;
+import tachyon.client.file.options.RenameOptions;
+import tachyon.client.file.options.SetAttributeOptions;
+import tachyon.client.file.options.UnmountOptions;
+import tachyon.exception.DirectoryNotEmptyException;
+import tachyon.exception.FileAlreadyExistsException;
+import tachyon.exception.FileDoesNotExistException;
+import tachyon.exception.InvalidPathException;
+import tachyon.exception.TachyonException;
+
+/**
+ * Basic file system interface supporting metadata operations and data operations. Developers
+ * should not implement this class but extend the default implementation provided by {@link
+ * BaseFileSystem} instead. This ensures any new methods added to the interface will be provided
+ * by the default implementation.
+ */
+@PublicApi
+interface FileSystem {
+  /**
+   * Convenience method for {@link #createDirectory(TachyonURI, CreateDirectoryOptions)} with
+   * default options.
+   *
+   * @param path the path of the directory to create in Tachyon space
+   * @return the {@link TachyonURI} referencing the newly created directory
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileAlreadyExistsException if there is already a file at the given path
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected exception is thrown
+   */
+  TachyonURI createDirectory(TachyonURI path)
```

Comment:
It's nice to have them in the interface so that users can work with `FileSystem`s instead of needing to use an abstract or concrete class to have access to convenience methods.